internal/expr/functions/math.go

```diff
@@ -30,3 +37,167 @@ var floorFunc = &ScalarDefinition{
 		}
 	},
 }
+
+var abs = &ScalarDefinition{
+	name:  "abs",
+	arity: 1,
+	callFn: func(args ...document.Value) (document.Value, error) {
+		switch args[0].Type {
```

Comment:
Ok 👍 makes sense to keep the same behaviour as the others, though those implicit casts feel so wrong 😆 

```go
v, err := args[0].CastAsDouble()
// if args[0] is NULL it will not return an error
```

I think there's a bug here, because it actually does!  https://github.com/genjidb/genji/blob/7a2a589d7861d095340a786e15669058c2a94182/document/cast.go#L103-L118

But the following code yields the expected result. 

```
genji> select CAST(NULL AS DOUBLE);
{
  "CAST(NULL AS double)": null
}
```

That's because the `NULL` handling is done elsewhere: https://github.com/genjidb/genji/blob/7a2a589d7861d095340a786e15669058c2a94182/document/cast.go#L16-L19

➡️ @asdine do you agree that it's a bug? Or is there a specific reason for `CastAs...` to not handle that case?